### ISSUE TITLE
Upgrade from `C123` -> `C124` on `Win x64` via `1.66.x - Nightly` (Checking C124)

### ISSUE THREAD

**Issue by kjozwiak** [brave/brave-browser#37383]
As per process, QA runs through the following cases to ensure that the major chromium bump that's about to get merged into `master` doesn't regress the `Nightly` channel. Once the major chromium bump is merged into `master`, QA will run through a full manual pass on `Windows` & `Android`.

### Startup & Components

- [x] Verify that Brave is only contacting `*.brave.com` endpoints on first launch using either `Charles Proxy`, `Fiddler`, `Wireshark` or `LittleSnitch` (or a similar application)
  - [x] Verify that opening a NTP doesn't trigger any outbound connections related to widgets without user interaction
- [x] Remove the following component folders and ensure that they're being re-downloaded after restarting the browser:
  - [x] `afalakplffnnnlkncjhbmahjfjhmlkal`: `AutoplayWhitelist.dat`, `ExtensionWhitelist.dat`, `ReferrerWhitelist.json` and `Greaselion.json`
  - [x] `CertificateRevocation`
  - [x] `gccbbckogglekeggclmmekihdgdpdgoe`: (Sponsored New Tab Images)
  - [x] `gkboaolpopklhgplhaaiboijnklogmbc`: Brave Ad Block List Catalog
  - [x] `iodkpdagapdfkphljnddpjlldadblomo`: Brave Ad Block Updater
  - [x] `mfddibmblmbccpadfndgakiopmmhebop`: Brave Ad Block List Catalog
  - [x] `oofiananboodjbbmdelgdommihjbkfag`: HTTPSE
  - [x] `Safe Browsing`
- [x] Restart the browser, load `brave://components`, wait for 8 mins and verify that no component shows any errors

**Note:** Always double check `brave://components` to make sure there's no errors/missing version numbers

### Upgrade

- [x] Make sure that data from the last version appears in the new version OK
- [x] Ensure that `brave://version` lists the expected Brave & Chromium versions
- [x] With data from the last version, verify that:
  - [x] Bookmarks on the bookmark toolbar and bookmark folders can be opened
  - [x] Cookies are preserved
  - [x] Installed extensions are retained and work correctly
  - [x] Opened tabs can be reloaded
  - [x] Stored passwords are preserved
  - [x] Sync chain created in previous version is retained
  - [x] Social media-blocking buttons changes are retained
  - [x] Custom filters under brave://settings/shields/filters are retained
  - [x] Custom lists under brave://settings/shields/filters are retained
  - [x] Rewards
    - [x] BAT balance is retained
    - [x] Auto-contribute list is retained
    - [x] Both Tips and Monthly Contributions are retained
    - [x] Panel transactions list is retained
    - [x] Changes to rewards settings are retained
    - [x] Ensure that Auto Contribute is not being enabled when upgrading to a new version if AC was disabled
  - [x] Ads
    - [x] Both `Estimated pending rewards` & `Ad notifications received this month` are retained
    - [x] Changes to ads settings are retained
    - [x] Ensure that ads are not being enabled when upgrading to a new version if they were disabled
    - [x] Ensure that ads are not disabled when upgrading to a new version if they were enabled

**Comment by MadhaviSeelam** [comment]
Verification `In progress` using
```
Brave | 1.66.59 Chromium: 124.0.6367.29 (Official Build) nightly (64-bit)
-- | --
Revision | 100c3c4b18cee34bbaff18754f86a7ff3867d707
OS | Windows 11 Version 23H2 (Build 22631.3374)
```

<details>
<summary>C123 --> C124 on-demand upgrade check - PASSED</summary>

1. installed latest nightly `1.66.67 Chromium 123.0.6312.105` and closed Brave
2. created `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
4. set `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x64` is used but this time around `nightly` was used)
5. relaunched Brave and visited `brave://settings/help` and ensured that new update was being pulled/downloaded
6. ensured that clicking `Relaunch` restarted the browser without any issues
7. ensured Brave was updated to `1.66.59 Chromium: 124.0.6367.29 (Official Build) nightly` without any issues and you can restart the browser

**`step 1`** | **`step 2`** | **`step 3`** | **`step 4`** | **`step 5`**
----------------|----------------|----------------|---------------|------------
![image](https://github.com/brave/brave-browser/assets/98358127/3497e4f9-507f-41d9-a977-2fc82dbbe45c)|![image](https://github.com/brave/brave-browser/assets/98358127/71b78820-3678-4ae8-9951-546204410263)|![image](https://github.com/brave/brave-browser/assets/98358127/fa650baf-6abc-4037-87e9-76ae30286c3c)|![image](https://github.com/brave/brave-browser/assets/98358127/5439ebff-0726-4947-8a6b-1390bd110e00)|![image](https://github.com/brave/brave-browser/assets/98358127/30d4cfcf-8e41-4be3-8802-582ce5331da4)


</details>

<details>
<summary>C123 --> C124 background upgrade check - PASSED</summary>

1. installed latest nightly `1.66.67 Chromium 123.0.6312.105` and closed Brave
2. created `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
3. set `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x64` is used but this time around `nightly` was used)
4. started `BraveSoftwareUpdateTaskMachineUA{E2D7A1CB-A35F-481A-AC5D-B3BDA377AD58}` via `Task Scheduler`
5. waited ~30 min or so before launching Brave and ensured it was updated to `1.66.59 Chromium: 124.0.6367.29 (Official Build) nightly`  
6. restarted Brave several times and ensured that there's no issues re-launching Brave once it was updated via the `background`

**`step 1`** | **`step 2`** | **`step 3`** | **`step 4`** | **`step 5`**
----------------|----------------|---------------|----------------|-----------
![image](https://github.com/brave/brave-browser/assets/98358127/6a29b5f6-92cd-4b61-9f3c-5fa62954fc37)|![image](https://github.com/brave/brave-browser/assets/98358127/bf3f7b99-3a25-41b0-a4e9-d3dd2fcf35a4)|![image](https://github.com/brave/brave-browser/assets/98358127/507c4a28-1e30-4dbd-a7c2-fb4d7b812f9c)|![image](https://github.com/brave/brave-browser/assets/98358127/6d7e1f34-6b39-413f-906e-7d921cc48d50)|![image](https://github.com/brave/brave-browser/assets/98358127/0fd5c84d-6217-4636-a6eb-5eaf9413d774)


</details>

<details>
<summary>C124 --> C124 on-demand upgrade check - Failed</summary>

* installed `1.66.59 Chromium: 124.0.6367.29` via `[BraveBrowserStandaloneNightlySetup_124_1_66_59.exe]
* created `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x64` is used but this time around `nightly` was used)
* relaunched Brave and visited `brave://settings/help` and ensured that new update was being pulled/downloaded
* ensured that clicking `Relaunch` restarted the browser without any issues
* ensured Brave was updated to `1.64.57 Chromium: 122.0.6261.29`  without any issues and you can restart the browser

**`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|----------------|---------------|----------
![image](https://github.com/brave/brave-browser/assets/98358127/614efb99-fd42-461a-90ef-424e24e0dba5)


</details>
<details>
<summary>C123 --> C123 background upgrade check - Pending</summary>

* installed `1.66.59 Chromium: 124.0.6367.29` via `[BraveBrowserStandaloneNightlySetup_124_1_66_59.exe]
* created `HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x86` is used but this time around `nightly` was used)
* started `BraveSoftwareUpdateTaskMachineUA{3841620D-F169-4927-BDD8-B34AB5FF24F1}` via `Task Scheduler`
* waited ~60min or so before launching Brave and ensured it was updated to `1.65.63 Chromium: 123.0.6312.29`
* restarted Brave several times and ensured that there's no issues re-launching Brave once it was updated via the `background

**`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|----------------|----------------|------------


</details>

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x86` using the following build(s):

```
Brave | 1.66.64 Chromium: 124.0.6367.29 (Official Build) nightly (32-bit)
-- | --
Revision | 3ca2bf9ba7a230733b7d089bd57eaf2dac350f85
OS | Windows 10 Version 22H2 (Build 19045.4239)
```

<details>
<summary>C123 --> C124 on-demand upgrade check - PASSED</summary>

* installed `1.66.56 Chromium: 123.0.6312.86` and closed Brave
* created `HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x86` is used but this time around `nightly` was used)
* realunched Brave and visited `brave://settings/help` and ensured that new update was being pulled/downloaded
* ensured that clicking `Relaunch` restarted the browser without any issues
* ensured Brave was updated to `1.66.64 Chromium: 124.0.6367.29` without any issues and you can restart the browser

**`Example`** | **`Example`** | **`Example`** | **`Example`** 
----------------|----------------|----------------|---------------
![image](https://github.com/brave/brave-browser/assets/2602313/f45c3f47-2a56-470f-ba25-7fd4a5f7d162) | ![image](https://github.com/brave/brave-browser/assets/2602313/4bc5827d-d758-4996-bf76-56ca15532374) | ![image](https://github.com/brave/brave-browser/assets/2602313/390d59ee-f531-49d0-8256-1df5be66063c) | ![image](https://github.com/brave/brave-browser/assets/2602313/2c574b9f-5ea1-4498-a0c0-df261cb9f661)

</details>

<details>
<summary>C124 --> C124 on-demand upgrade check - PASSED</summary>

* installed `1.66.59 Chromium: 124.0.6367.29` via `BraveBrowserStandaloneNightlySetup32_124_1_66_59.exe` and closed Brave
* created `HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x86` is used but this time around `nightly` was used)
* realunched Brave and visited `brave://settings/help` and ensured that new update was being pulled/downloaded
* ensured that clicking `Relaunch` restarted the browser without any issues
* ensured Brave was updated to `1.66.64 Chromium: 124.0.6367.29` without any issues and you can restart the browser

**`Example`** | **`Example`** | **`Example`** | **`Example`** 
----------------|----------------|----------------|----------------
![image](https://github.com/brave/brave-browser/assets/2602313/cad8382a-b7e9-4f30-9a4a-d933f6dd49ee) | ![image](https://github.com/brave/brave-browser/assets/2602313/dd9ca810-e8a2-4c22-8b78-23d5485f2b98) | ![image](https://github.com/brave/brave-browser/assets/2602313/e5dc13c0-a181-4458-b8b6-e5afeccf7190) | ![image](https://github.com/brave/brave-browser/assets/2602313/ec0f8caf-1045-4b1c-bc4e-1788d109a456)

`x86` delta logs from `C:\ProgramData\BraveSoftware\Update\Log\BraveUpdate.log`:

```
[04/08/24 16:51:19.127][BraveUpdate:goopdate][6988:8760][Installing][display name: Brave Nightly][app id: {C6CB981E-DB30-4876-8639-109F8933582C}][installer path: C:\Program Files\BraveSoftware\Update\Install\{8CA174C2-0517-4043-B3A8-AB33FF83219C}\brave_installer-delta-ia32.exe][manifest args: --chrome-sxs][installer data: ][untrusted data: ][priority: 10]
[04/08/24 16:51:19.128][BraveUpdate:goopdate][6988:8760][Running installer][C:\Program Files\BraveSoftware\Update\Install\{8CA174C2-0517-4043-B3A8-AB33FF83219C}\brave_installer-delta-ia32.exe][--chrome-sxs][{C6CB981E-DB30-4876-8639-109F8933582C}]
[04/08/24 16:51:35.792][BraveUpdate:goopdate][6988:8760][InstallerResult][{C6CB981E-DB30-4876-8639-109F8933582C}][0]
[04/08/24 16:51:35.792][BraveUpdate:goopdate][6988:8760][InstallApp returned][0x0][{C6CB981E-DB30-4876-8639-109F8933582C}][type:1][code: 30][][]
[04/08/24 16:51:35.796][BraveUpdate:goopdate][6988:8760][Started process][6516]
[04/08/24 16:51:35.798][BraveUpdate:goopdate][6988:8760][LogEventHelper][4][12][1][[
Application update/install.
pid=6988, ver=1.3.361.149, machine=1, extern=1, private
App={C6CB981E-DB30-4876-8639-109F8933582C}, Ver=124.1.66.59, Download result=0x00000000
App={C6CB981E-DB30-4876-8639-109F8933582C}, Ver=124.1.66.59, Install result=0x00000000
]][]
[04/08/24 16:51:35.803][BraveCrashHandler][6516:6628][ConfigManager::LoadGroupPolicies][Machine is not Enterprise Managed]
[04/08/24 16:51:35.803][BraveCrashHandler][6516:6628][ConfigManager::LoadGroupPolicies][No Group Policies found under key][HKLM\Software\Policies\BraveSoftware\Update\]
[04/08/24 16:51:35.803][BraveCrashHandler][6516:6628][OmahaPolicyManager::set_policy][Group Policy][[CachedOmahaPolicy][is_initialized][0][is_managed][0][auto_update_check_period_minutes][-1][download_preference][][cache_size_limit][-1][cache_life_limit][-1][updates_suppressed][-1][-1][-1][proxy_mode][][proxy_server][][proxy_pac_url][][install_default][-1][update_default][-1]]
[04/08/24 16:51:35.805][BraveCrashHandler][6516:6628][CrashHandler::RunUntilShutdown]
[04/08/24 16:51:35.901][BraveUpdate:goopdate][5372:7024][DllEntry exit][0x00000000]
```

</details>

<details>
<summary>C123 --> C124 background upgrade check - PASSED</summary>

* installed `1.66.56 Chromium: 123.0.6312.86` and closed Brave
* created `HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x86` is used but this time around `nightly` was used)
* started `BraveSoftwareUpdateTaskMachineUA{4D9A5BA7-32E9-4B1C-8F65-AAF50E1E3CD3}` via `Task Scheduler`
* waited ~30min or so before launching Brave and ensured it was updated to `1.66.64 Chromium: 124.0.6367.29`
* restarted Brave several times and ensured that there's no issues re-launching Brave once it was updated via the `background`

**`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|---------------|-----------------
![image](https://github.com/brave/brave-browser/assets/2602313/5107d8f1-305d-4379-a8c2-c326e0926e51) | ![image](https://github.com/brave/brave-browser/assets/2602313/c16eb00e-f96d-4c37-88b7-c0fc5120141f) | ![image](https://github.com/brave/brave-browser/assets/2602313/b1cd16bc-8d73-4463-90c7-f92b591dad1a) | ![image](https://github.com/brave/brave-browser/assets/2602313/703936d9-6280-42bc-b7e5-09f67f6408c9)

</details>

<details>
<summary>C124 --> C124 background upgrade check - PASSED</summary>

* installed `1.66.59 Chromium: 124.0.6367.29` via `BraveBrowserStandaloneNightlySetup32_124_1_66_59.exe` and closed Brave
* created `HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x86` is used but this time around `nightly` was used)
* started `BraveSoftwareUpdateTaskMachineUA{0CB2B8C6-95E1-42D7-99AE-FA9E276ECCE9}` via `Task Scheduler`
* waited ~30min or so before launching Brave and ensured it was updated to `1.66.64 Chromium: 124.0.6367.29`
* restarted Brave several times and ensured that there's no issues re-launching Brave once it was updated via the `background

**`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|----------------|----------------
![image](https://github.com/brave/brave-browser/assets/2602313/04e045a3-316a-4280-84da-3cc6a11ffa5a) | ![image](https://github.com/brave/brave-browser/assets/2602313/3458fbce-35ef-478a-a1c1-b79ecd3b85cf) | ![image](https://github.com/brave/brave-browser/assets/2602313/3fc2e25f-a59e-475f-884d-71e7761522bc) | ![image](https://github.com/brave/brave-browser/assets/2602313/bafcb5ce-b9a2-4da9-9cf8-e5374d82b258)

`x86` delta logs from `C:\ProgramData\BraveSoftware\Update\Log\BraveUpdate.log`:

```
[04/08/24 17:18:37.378][BraveUpdate:goopdate][11048:11652][Installing][display name: Brave Nightly][app id: {C6CB981E-DB30-4876-8639-109F8933582C}][installer path: C:\Program Files\BraveSoftware\Update\Install\{02285CD2-7323-44DF-A13B-7A71EC16F692}\brave_installer-delta-ia32.exe][manifest args: --chrome-sxs][installer data: ][untrusted data: ][priority: 10]
[04/08/24 17:18:37.378][BraveUpdate:goopdate][11048:11652][Running installer][C:\Program Files\BraveSoftware\Update\Install\{02285CD2-7323-44DF-A13B-7A71EC16F692}\brave_installer-delta-ia32.exe][--chrome-sxs][{C6CB981E-DB30-4876-8639-109F8933582C}]
[04/08/24 17:18:54.689][BraveUpdate:goopdate][11048:11652][InstallerResult][{C6CB981E-DB30-4876-8639-109F8933582C}][0]
[04/08/24 17:18:54.689][BraveUpdate:goopdate][11048:11652][InstallApp returned][0x0][{C6CB981E-DB30-4876-8639-109F8933582C}][type:1][code: 30][][]
[04/08/24 17:18:54.700][BraveUpdate:goopdate][11048:11652][Started process][9564]
[04/08/24 17:18:54.702][BraveUpdate:goopdate][11048:11652][LogEventHelper][4][12][1][[
Application update/install.
pid=11048, ver=1.3.361.149, machine=1, extern=1, private
App={C6CB981E-DB30-4876-8639-109F8933582C}, Ver=124.1.66.59, Download result=0x00000000
App={C6CB981E-DB30-4876-8639-109F8933582C}, Ver=124.1.66.59, Install result=0x00000000
]][]
[04/08/24 17:18:54.710][BraveCrashHandler][9564:9860][ConfigManager::LoadGroupPolicies][Machine is not Enterprise Managed]
[04/08/24 17:18:54.710][BraveCrashHandler][9564:9860][ConfigManager::LoadGroupPolicies][No Group Policies found under key][HKLM\Software\Policies\BraveSoftware\Update\]
[04/08/24 17:18:54.710][BraveCrashHandler][9564:9860][OmahaPolicyManager::set_policy][Group Policy][[CachedOmahaPolicy][is_initialized][0][is_managed][0][auto_update_check_period_minutes][-1][download_preference][][cache_size_limit][-1][cache_life_limit][-1][updates_suppressed][-1][-1][-1][proxy_mode][][proxy_server][][proxy_pac_url][][install_default][-1][update_default][-1]]
[04/08/24 17:18:54.713][BraveCrashHandler][9564:9860][CrashHandler::RunUntilShutdown]
[04/08/24 17:18:54.801][BraveUpdate:goopdate][11048:6628][Started process][12188]
[04/08/24 17:18:54.803][BraveUpdate:goopdate][7772:7372][DllEntry exit][0x00000000]
```

</details>
